projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -327,11 +328,44 @@ export class IgxOverlayService implements OnDestroy {
             console.warn('Overlay was not able to attach provided component!');
             return null;
         }
+        
+        settings = Object.assign({}, this._defaultSettings, settings);
+
+        let target = {
```

Comment:
No need of the `target`. All the properties and methods added in the `target` should be added in the handler.

Actually the `settings` is the target. If we set `info.settings = new Proxy(target, handler);` anyone having access to the `info.settings` will not receive `OverlaySettings` interface but some strange object, which is not ok.